src/main/java/duke/tasks/Deadline.java

```diff
@@ -0,0 +1,16 @@
+package duke.tasks;
+
+public class Deadline extends Task {
+
+    protected String end;
+    public Deadline(String description, String end)
+    {
+        super(description);
+        this.end = end;
+    }
+
+    @Override
+    public void printTask() {
+        System.out.println("[D][" + getStatusIcon() + "] " + description + "(by:" + end + ")");
```

Comment:
Arguably, strings like `[D]` can be seen as a magic strings. You might want to consider extracting strings like this into a named constants. Sometimes, you could also use enums to achieve similar effect. There could be other magic strings/numbers/literals in your other code too that you might want to look out for them.